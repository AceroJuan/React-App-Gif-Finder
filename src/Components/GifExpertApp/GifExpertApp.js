import React, { useState } from "react";
import AddCategory from "../AddCategory/AddCategory";
import GifGrid from "../GifGrid/GifGrid";
import "animate.css";

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h1 className="animate__animated animate__fadeIn">Gif Expert App</h1>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ul>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ul>
    </>
  );
};

export default GifExpertApp;
