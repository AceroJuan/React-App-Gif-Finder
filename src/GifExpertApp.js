import React, { useState } from "react";
import PropTypes from "prop-types";
import AddCategory from "./Components/addCategory/AddCategory";

const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    "one punch",
    "samurai x",
    "dragon ball",
  ]);

  //   const handleAdd = () => {
  //     // let newCategory = "sailor moon";
  //     // setCategories(categories.concat(newCategory));
  //     setCategories(["slam dunk", ...categories]);
  //   };

  return (
    <>
      <main>
        <h1>GifExpertApp</h1>
        <AddCategory />
        <hr />

        <ul>
          {categories.map((category) => {
            return <li key={category}>{category}</li>;
          })}
        </ul>
      </main>
    </>
  );
};

GifExpertApp.propTypes = {};

export default GifExpertApp;
