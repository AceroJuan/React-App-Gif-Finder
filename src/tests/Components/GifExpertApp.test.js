import React from "react";
import "@testing-library/jest-dom";
import GifExpertApp from "../../Components/GifExpertApp/GifExpertApp";
import { shallow } from "enzyme";

describe("pruebas en GifExpertApp", () => {
  test("debe mostrar <GifExpertApp /> de manera correcta", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
});
