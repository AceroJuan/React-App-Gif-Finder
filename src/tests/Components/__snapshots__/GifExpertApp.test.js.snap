// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifExpertApp debe mostrar <GifExpertApp /> de manera correcta 1`] = `
<Fragment>
  <h1
    className="animate__animated animate__fadeIn"
  >
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category=""
      key=""
    />
  </ul>
</Fragment>
`;
